cli/internal/sync/map.go

```diff
@@ -0,0 +1,63 @@
+package sync
```

Comment:
Why have this instead of sync.Map? 🤔

Reply:
because I want to be type safe. I hate that sync.Map always needs type assertion

Reply:
Fair enough.